### ISSUE TITLE
`jax.nn.sigmoid` default JVP loses a finite float64 tail derivative that JAX's highest-accuracy rule preserves

### ISSUE THREAD

**Issue by beanduan22** [jax-ml/jax#40342]
### Description

`jax.grad(jax.nn.sigmoid)` returns `0.0` at a finite positive `float64` input where the derivative is finite and normal. At `x = 40.0`, the expected derivative is `4.248354255291589e-18`, but the default public API returns zero.

The negative-side derivative at `x = -40.0` is retained, so the result is not caused by underflow. JAX's own `lax.logistic` rule with `accuracy=lax.AccuracyMode.HIGHEST` returns the expected nonzero derivative at the same input, demonstrating that a stable derivative path exists.

```python
import math
import jax
import jax.numpy as jnp
from jax import lax

jax.config.update("jax_enable_x64", True)
x_positive = jnp.asarray(40.0, dtype=jnp.float64)
x_negative = jnp.asarray(-40.0, dtype=jnp.float64)
default_grad = jax.grad(jax.nn.sigmoid)
highest_grad = jax.grad(lambda x: lax.logistic(x, accuracy=lax.AccuracyMode.HIGHEST))
print("default grad(+40):", float(default_grad(x_positive)))
print("default grad(-40):", float(default_grad(x_negative)))
print("highest-accuracy grad(+40):", float(highest_grad(x_positive)))
print("expected:", math.exp(-40.0) / (1.0 + math.exp(-40.0)) ** 2)
```

## Actual result

On JAX `0.9.2`, CPU and two CUDA devices reproduce:

```text
default grad(+40): 0.0
default grad(-40): 4.24835425529159e-18
highest-accuracy grad(+40): 4.24835425529159e-18
expected: 4.248354255291589e-18
```

## Expected result

`jax.grad(jax.nn.sigmoid)(40.0)` should remain finite and close to `4.248354255291589e-18`, matching the highest-accuracy derivative path.

### System info (python version, jaxlib version, accelerator, etc.)

```text
JAX version: 0.9.2
jaxlib version: 0.9.2
Python version: 3.13.5
Input dtype: float64, with jax_enable_x64=True
Devices tested: CPU and two CUDA GPUs
```

**Comment by jakevdp** [comment]
Thanks for the report – this looks to fall under https://github.com/jax-ml/jax/issues/38651.
